pep-3119.txt

```diff
@@ -844,65 +842,54 @@ References
 .. [1] An Introduction to ABC's, by Talin
    (https://mail.python.org/pipermail/python-3000/2007-April/006614.html)
 
-.. [2] Incomplete implementation prototype, by GvR
-   (http://svn.python.org/view/sandbox/trunk/abc/)
+[2] Incomplete implementation prototype, by GvR
+\   (https://web.archive.org/web/20170223133820/http://svn.python.org/view/sandbox/trunk/abc/)
 
-.. [3] Possible Python 3K Class Tree?, wiki page created by Bill Janssen
-   (http://wiki.python.org/moin/AbstractBaseClasses)
+[3] Possible Python 3K Class Tree?, wiki page created by Bill Janssen
+\   (https://wiki.python.org/moin/AbstractBaseClasses)
 
 .. [4] Generic Functions implementation, by GvR
-   (http://svn.python.org/view/sandbox/trunk/overload/)
+   (https://web.archive.org/web/20170223135019/http://svn.python.org/view/sandbox/trunk/overload/)
 
 .. [5] Charming Python: Scaling a new PEAK, by David Mertz
-   (http://www-128.ibm.com/developerworks/library/l-cppeak2/)
+   (https://web.archive.org/web/20070515125102/http://www-128.ibm.com/developerworks/library/l-cppeak2/)
 
 .. [6] Implementation of @abstractmethod
-   (https://bugs.python.org/issue1706989)
+   (https://github.com/python/cpython/issues/44895)
 
 .. [7] Unifying types and classes in Python 2.2, by GvR
-   (http://www.python.org/download/releases/2.2.3/descrintro/)
+   (https://www.python.org/download/releases/2.2.3/descrintro/)
 
 .. [8] Putting Metaclasses to Work: A New Dimension in Object-Oriented
    Programming, by Ira R. Forman and Scott H. Danforth
-   (http://www.amazon.com/gp/product/0201433052)
+   (https://archive.org/details/PuttingMetaclassesToWork)
```

Comment:
Whoa, this link works. Is archive.org special-casing certain books?

Reply:
I was also amazed that the Amazon.com link works some 16 years on!

Reply:
So it appears archive.org lost a lawsuit about their hosting free copies of books[1]. This link (and others like it) might stop working as a result?

___
[1]:
https://www.nytimes.com/2023/08/13/business/media/internet-archive-emergency-lending-library.html?unlocked_article_code=oN9FPAI5vRRvlOHExHeICsjbWeLOb9yW8BFOGpV4LXhbA1LdFbZh5wToe2Zlnk7ZMw9QmreqdIiZ9MA71rXiPOW7IntD2HO93EESuSwHK0XVeVnAIuBuCyu1oV3wB1NvhXvQ4GRoZw3oQm0HAVUsJkU1Ol2GkHJlC2LoWC_kvpWo4fIAzUkcyu68ZKWIB_jeQoZoGYnxlsXYnlp6FJLCI-l6fFCzC607dxWtaru8k3PVPqlxr8E1eWZgz6vG3tORGQpBcLPeYudZfIWEc97KcqybsrgHIJkQoSV_kIyerVaqyfsgGlGHu2ptguGG2bsD7bfrXLGbrbCHxeoCNPfPvZb5DN81Eq2FXLgrW6kukTuEpqLuti2bqvxsFw&smid=url-share

Reply:
Thanks for letting me know, we probably shouldn't link to legally dubious content -- I have opened #3289 to go back to the old link.

A